pkg/frontend/middleware/authenticated_test.go

```diff
@@ -0,0 +1,108 @@
+package middleware
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/gorilla/mux"
+
+	"github.com/Azure/ARO-RP/pkg/util/clientauthorizer"
+	mock_env "github.com/Azure/ARO-RP/pkg/util/mocks/env"
+)
+
+func TestAuthenticatedForOCMAPIs(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+	_env := mock_env.NewMockInterface(controller)
+	r := mux.NewRouter()
+	// mimic what the setupRouter func will do for this specific path
+	r.HandleFunc("/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}",
+		func(w http.ResponseWriter, request *http.Request) {})
+
+	r.HandleFunc("/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}/{ocmResourceType}/{ocmResourceName}",
+		func(w http.ResponseWriter, request *http.Request) {}).
+		Queries("api-version", "{api-version}")
+	basePath := "https://server/subscriptions/0000-0000/resourcegroups/testrg/providers/testrpn/testrt/testrn/%s/myResource?api-version=2022-09-04"
+
+	for _, tt := range []struct {
+		name             string
+		method           string
+		ocmResourceType  string
+		systemDataHeader string
+		wantStatus       int
+		isValid          bool
+		mocks            func(_env *mock_env.MockInterface, req http.Request, isValid bool)
+	}{
+		{
+			name:             "non ocm api called, system data header is not validated",
+			method:           "GET",
+			ocmResourceType:  "",
+			systemDataHeader: `{"systemData":{"lastModifiedBy":"unused"}}`,
+			wantStatus:       200,
+			isValid:          true,
+			mocks: func(_env *mock_env.MockInterface, req http.Request, isValid bool) {
+				_env.EXPECT().ArmClientAuthorizer().Return(clientauthorizer.NewAll())
+				_env.EXPECT().ValidateOCMClientID(req.Header.Get(ArmSystemDataHeaderKey)).Times(0)
+			},
```

Comment:
these funcs as variables aren't great. We could maybe just get away with having a `expectCalled` `expectArg` and `return` args so we would not have duplication of `_env.EXPECT().ArmClientAuthorizer().Return(clientauthorizer.NewAll())` and no func vars

Reply:
yeah I can refactor the test to remove the function variables